src/components/weather-widget/WeatherWidget.vue

```diff
@@ -0,0 +1,235 @@
+<template>
+  <div class="weather-widget-wrapper d-flex flex-column align-items-center">    
+    <div class="weather-search d-flex justify-content-center">
+      <input 
+        v-model="city" 
+        class="search-city" 
+        type="text" 
+        placeholder="Search city...">
+      <button 
+        @click="getWeather">
+        &#128269;
+      </button>
+    </div>
+    <div 
+      v-if="searched" 
+      class="weather-information d-flex flex-column align-items-center">
+      <div class="weather-statement">
+        {{ weather.statement }}
+        <img 
+          :src="weather.icon">
+      </div>
+      <div class="weather-temperature d-flex justify-content-center align-items-center">
+        {{ weather.temperature }}
+      </div>
+      <div class="weather-min-max-temp d-flex">
+        <div class="weather-min-temperature d-flex justify-content-center align-items-center">
+          {{ weather.minTemperature }}
+          <i class="fas fa-temperature-low"></i>
+        </div>
+        <div class="weather-max-temperature d-flex justify-content-center align-items-center">
+          {{ weather.maxTemperature }}
+          <i class="fas fa-temperature-high"></i>
+        </div>
+      </div>
+      <div class="weather-additional-info d-flex">
+        <div class="weather-location-info text-center d-flex flex-column justify-content-center align-items-center">
+          <i class="fas fa-map-marker-alt"></i>
+          <p>{{ weather.localInfo }}</p>
+        </div>
+        <div class="weather-humidity d-flex justify-content-center align-items-center">
+          {{ weather.humidity }}
+          <i class="fas fa-tint"></i>
+        </div>
+      </div>
+    </div>
+    <div 
+      v-else 
+      class="weather-initial-message d-flex align-items-center text-center">
+      Please, enter correct city🌞
+    </div>
+  </div>
+</template>
+
+<script>
+  export default {
+    data() {
+      return {
+        API_KEY: 'e4ca8e74b9ce1d7d81d543944f39117c',
```

Comment:
please, move api token to constants file

Reply:
Ok!